plugin/builtin/patchstrategicmergetransformer/PatchStrategicMergeTransformer_test.go

```diff
@@ -1060,43 +1144,39 @@ func TestMultiplePatches(t *testing.T) {
 		},
 	}
 
+	if konfig.FlagEnableKyamlDefaultValue {
```

Comment:
Can we add something like `disableForKyaml` in `testRecord` so that we don't need to have this hardcoded list? If this is intended to be removed in near future, please add a TODO.

Reply:
Added todo.  I like the idea about adding the field, but we still need an if statement, and this collects all the work in one list.  The goal is to get these tests to pass, on consciously - with reasoning - delete them from the code base - in subsequent prs.

If you don't like this approach, i can withdraw all these if statements.  I now know the bounds of the problem.  but i though that by doing it this way, others could help work on it, because it's all clearly labelled